libs/core/src/main/java/org/opensearch/core/transport/TransportMessage.java

```diff
@@ -53,6 +56,10 @@ public TransportAddress remoteAddress() {
         return remoteAddress;
     }
 
+    public boolean isMessageProtobuf() {
```

Comment:
This feels like an antipattern. Should we introduce a message type and make it an enum?

Reply:
Fixed.